### PR TITLE
upgrade go to 1.21 and fix deps version

### PR DESCRIPTION
**Pull request by fahedouch**
### Description of your changes

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,27 +14,27 @@ PLATFORMS ?= linux_amd64 linux_arm64
 # ====================================================================================
 # Setup Go
 
-GO_REQUIRED_VERSION = 1.19
+#GO_REQUIRED_VERSION = 1.21
```
Comment (d-honeybadger) [inline Makefile:17]: Any particular reason for not removing these commented out versions?

**Review by d-honeybadger — APPROVED** [review]
I'm not well versed in the details of crossplane pkg updates, so take it with a grain of salt,
but I just deployed this branch in my cluster and it appears to be working, hence the approval

File: package/crds/ansible.crossplane.io_ansibleruns.yaml [hunk 2]
```diff
@@ -41,9 +40,13 @@ spec:
             properties:
               deletionPolicy:
                 default: Delete
-                description: DeletionPolicy specifies what will happen to the underlying
+                description: 'DeletionPolicy specifies what will happen to the underlying
```
Comment (d-honeybadger) [inline package/crds/ansible.crossplane.io_ansibleruns.yaml:43]: Are these changes brought in automagically with crossplane upgrade?

Comment (fahedouch) [reply package/crds/ansible.crossplane.io_ansibleruns.yaml:43]: Yes

File: Makefile (hunk 1 above)
Comment (fahedouch) [reply Makefile]: it was for testing reasons, fixed

**Pull request merged**